jfr-streaming/src/main/java/io/opentelemetry/contrib/jfr/metrics/internal/GarbageCollection/OldGarbageCollectionHandler.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.jfr.metrics.internal.GarbageCollection;
+
+import static io.opentelemetry.contrib.jfr.metrics.internal.Constants.ATTR_ACTION;
+import static io.opentelemetry.contrib.jfr.metrics.internal.Constants.ATTR_GC;
+import static io.opentelemetry.contrib.jfr.metrics.internal.Constants.DURATION;
+import static io.opentelemetry.contrib.jfr.metrics.internal.Constants.END_OF_MAJOR_GC;
+import static io.opentelemetry.contrib.jfr.metrics.internal.Constants.METRIC_DESCRIPTION_GC_DURATION;
+import static io.opentelemetry.contrib.jfr.metrics.internal.Constants.METRIC_NAME_GC_DURATION;
+import static io.opentelemetry.contrib.jfr.metrics.internal.Constants.MILLISECONDS;
+import static io.opentelemetry.contrib.jfr.metrics.internal.RecordedEventHandler.defaultMeter;
+
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.metrics.LongHistogram;
+import io.opentelemetry.api.metrics.Meter;
+import io.opentelemetry.contrib.jfr.metrics.internal.RecordedEventHandler;
+import java.time.Duration;
+import java.util.HashSet;
+import java.util.Optional;
+import jdk.jfr.consumer.RecordedEvent;
+
+public final class OldGarbageCollectionHandler implements RecordedEventHandler {
+  private final HashSet<String> garbageCollectors;
+  private static final String EVENT_NAME = "jdk.OldGarbageCollection";
+  // This could be changed later
+  private Attributes attributes =
+      Attributes.of(ATTR_GC, "MarkSweepCompact", ATTR_ACTION, END_OF_MAJOR_GC);
+
+  private LongHistogram histogram;
+
+  public OldGarbageCollectionHandler(HashSet<String> garbageCollectors) {
```

Comment:
`Set<String>` is preferable to `HashSet<String>` because this code doesn't care which implementation of `Set` is used.